### DISCUSSION TITLE
Headless event hooks

### DISCUSSION THREAD

**Discussion by joshlarsen** [projectdiscovery/nuclei#1793] [Q&A] [answered]
Is there a way to hook into or respond to browser events in headless mode? Similar to the way Puppeteer allows you to hook events like `page.on('response')` for the response event or `client.on('Network.webSocketCreated')` when a websocket connection is opened.

**Comment by forgedhallpass** [comment, marked as answer]
Hello @joshlarsen,

Have you tried/seen the [`waitevent`](https://github.com/projectdiscovery/nuclei/blob/54d8f9fdb5a5022f1abe812b9a340d100e3082f1/v2/pkg/protocols/headless/engine/page_actions.go#L506) action, which expects an `event` and an optional `timeout` parameter?

p.s. Nuclei delegates to the [go-rod](https://github.com/go-rod/rod) library for headless interactions, hence if needed we could potentially extend the nuclei headless API to include more functionality.

**Reply by joshlarsen** [reply]
I missed that somehow, thanks!
